OSDOCS-9919 OCP z-stream RNs 4.15.2 (#72951)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-15-2"]
=== RHSA-2024:1210 - {product-title} 4.15.2 bug fix and security update

Issued: 2024-3-13

> {product-title} release {product-version}.0, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:1210[RHSA-2024:1210] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:1213[RHBA-2024:1213] advisory.

Reviewer: `.2`:
```suggestion
{product-title} release {product-version}.2, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:1210[RHSA-2024:1210] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:1213[RHBA-2024:1213] advisory.
```

Author: Good catch, thanks!

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.2 --pullspecs
----

[id="ocp-4-15-2-known-issues"]
==== Known issues

> * Providing a performance profile as an extra manifest at Day 0 did not work in 4.15.0 but is now possible in 4.15.2 with the following limitation.
> +
> The installation of OCP might fail when a performance profile is present in the extra manifests folder and targets the primary or worker pools. This is caused by the internal install ordering that processes the performance profile before the default primary and worker `MachineConfigPools` are created. It is possible to workaround this issue by including a copy of the stock primary or worker `MachineConfigPools` in the extra manifests folder. (link:https://issues.redhat.com/browse/OCPBUGS-27948[*OCPBUGS-27948*], link:https://issues.redhat.com/browse/OCPBUGS-29752[*OCPBUGS-29752*])

Reviewer: ```suggestion
* Providing a performance profile as an extra manifest at Day 0 did not work in {product-title} 4.15.0, but it is now possible in 4.15.2 with the following limitation:
+
The installation of {product-title} might fail when a performance profile is present in the extra manifests folder and targets the primary or worker pools. This is caused by the internal installation ordering that processes the performance profile before the default primary and worker `MachineConfigPools` are created. It is possible to workaround this issue by including a copy of the stock primary or worker `MachineConfigPools` in the extra manifests folder. (link:https://issues.redhat.com/browse/OCPBUGS-27948[*OCPBUGS-27948*], link:https://issues.redhat.com/browse/OCPBUGS-29752[*OCPBUGS-29752*])
```

[id="ocp-4-15-2-bug-fixes"]
==== Bug fixes

> * Previously, when updating to {product-title} 4.15 the CatalogSource never refreshed which caused the optional operators to fail to update. With this release, the image pull policy is changed to `Always` which enables the optional operators to update correctly. (link:https://issues.redhat.com/browse/OCPBUGS-30193[*OCPBUGS-30193*])

Reviewer: ```suggestion
* Previously, when updating to {product-title} 4.15, `CatalogSource` objects never refreshed, which caused the optional Operator catalogs to fail to update. With this release, the image pull policy is changed to `Always`, which enables the optional Operator catalogs to update correctly. (link:https://issues.redhat.com/browse/OCPBUGS-30193[*OCPBUGS-30193*])
```

> * Previously, the `nodeStatusReportFrequency` was linked to the `nodeStatusUpdateFrequency`. With this release, the `nodeStatusReportFrequency` is set to 5 minutes. (link:https://issues.redhat.com/browse/OCPBUGS-29797[*OCPBUGS-29797*])

Reviewer: ```suggestion
* Previously, the `nodeStatusReportFrequency` setting was linked to the `nodeStatusUpdateFrequency` setting. With this release, the `nodeStatusReportFrequency` setting is set to 5 minutes. (link:https://issues.redhat.com/browse/OCPBUGS-29797[*OCPBUGS-29797*])
```

> * Previously, under certain conditions the Installer would fail with the error message: `unexpected end of JSON input`. With this release, the error message is clarified and suggests users set the `serviceAccount` field in the install-config.yaml to fix the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29495[*OCPBUGS-29495*])

Reviewer: ```suggestion
* Previously, under certain conditions, the installer would fail with the error message `unexpected end of JSON input`. With this release, the error message is clarified and suggests users set the `serviceAccount` field in the `install-config.yaml` file to fix the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29495[*OCPBUGS-29495*])
```

> * Previously, the oauthMetadata provided in the HostedCluster was not honored. With this release, the oauthMetadata is honored by the HostedCluster. (link:https://issues.redhat.com/browse/OCPBUGS-29025[*OCPBUGS-29025*])

Reviewer: ```suggestion
* Previously, the `oauthMetadata` property provided in the `HostedCluster` object was not honored. With this release, the `oauthMetadata` property is honored by the `HostedCluster` object. (link:https://issues.redhat.com/browse/OCPBUGS-29025[*OCPBUGS-29025*])
```
